.ci/releasenotes-gen-xdoc-push.sh

```diff
@@ -39,7 +45,7 @@ java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.
      -localRepoPath checkstyle \
      -remoteRepoPath checkstyle/checkstyle \
      -startRef "$LATEST_RELEASE_TAG" \
-     -releaseNumber "$CS_RELEASE_VERSION" \
+     -releaseNumber "$TARGET_VERSION" \
```

Comment:
Just curious why this wasn't an issue before? I checked and the last release did not update the POM until after the release notes were committed.

Reply:
We never used this script before
We always use
https://github.com/checkstyle/checkstyle/actions/runs/3588930515/jobs/6040835341#step:3:1
`.ci/releasenotes-gen.sh`

Reply:
Then are we fixing the right script or was this script, which was never used before, used by accident or just that new?

Reply:
We have two scripts that are similar: for release action and for validation action.
We can to think later is we need to optimize them to use one

Reply:
Then can we create an issue so its not forgotten?

Reply:
I am not sure it makes sense. I waiting for final completion of auto release project, to rethink something at end. To clearly see what we need and what is convenient.